reader.go

```diff
@@ -137,3 +137,58 @@ func (r *Reader) ensureOpen() error {
 
 	return nil
 }
+
+// -----------------------------------------------------------------------------------------------
+
+// ReaderIter is a reader iterator.
+type ReaderIter struct {
+	remotes []*bfs.Object
+	opt     *ReaderOptions
+	ctx     context.Context
+	reader  *Reader
+
+	i   int
+	err error
+	num int
+}
+
+// NewReaderIter inits a ReaderIter.
+func NewReaderIter(ctx context.Context, remotes []*bfs.Object, opt *ReaderOptions) *ReaderIter {
```

Comment:
hmmm, i'm trying to remember why I added this over just passing a slice of readers...  I think it was prob so that caller handles initing/closing readers rather than having to open and defer closing a whole group of readers on consumer side.

Reply:
Please remove. This is not something the user should be aware of. All they want is iterate over records. The way to solve this nicely is not simple though:

1. You could make `type Reader struct` a private `type streamReader struct`
2. Add a new `type Reader struct { remotes []*bfs.Object; pos int; cur *streamReader }` which then iterates over remotes and secretly creates new `streamReader` for each.